lib/secure-hash.js

```diff
@@ -0,0 +1,39 @@
+"use strict";
+
+module.exports = function (base64, crypto, promise) {
+    crypto.pbkdf2Async = promise.promisify(crypto.pbkdf2);
+
+    /**
+     * Hashes the content passed using defaults or
+     * configuration options if passed in.
+     *
+     * @param {(Buffer|string)} data
+     * @param {Object} config
+     * @return {Promise.<string>}
+     */
+    function secureHashAsync(data, config) {
+        var algorithm, hashLength, iterations, salt;
+
+        if (! data) {
+            throw new Error("Nothing to hash");
+        }
+
+        if (typeof data === "string") {
+            data = new Buffer(data, "binary");
+        }
+
+        config = config || {};
+        algorithm = config.algo || "sha512";
+        hashLength = config.hashLength || 512;
```

Comment:
Lower this considerably.  How about 48 for a total of 64 characters once encoded?

4 * bytesRaw / 3 = bytesEncoded.  33.33333333333333333% (roughly) increase in size.